source/common/upstream/upstream_impl.h

```diff
@@ -508,6 +508,8 @@ class ClusterInfoImpl : public ClusterInfo {
 
   bool drainConnectionsOnHostRemoval() const override { return drain_connections_on_host_removal_; }
 
+  absl::optional<std::string> eds_service_name() const override { return eds_service_name_; }
```

Comment:
I think `eds_service_name` is fine here.

Reply:
I am not sure I understand you here. Are you talking about the name or the return value type?

Reply:
I was just pointing out that it's fine to use this internally, but not really the best choice in API. Resolved.